router/usecase/router_usecase.go

```diff
@@ -345,37 +349,6 @@ func (r *routerUseCaseImpl) computeAndRankRoutesByDirectQuote(ctx context.Contex
 	return topSingleRouteQuote, rankedRoutes, nil
 }
 
-// estimateDirectQuote estimates and returns the direct quote for the given routes, token in and token out denom.
-// Also, returns the routes ranked by amount out in decreasing order.
-// Returns error if:
-// - fails to estimate direct quotes
-func estimateDirectQuote(ctx context.Context, routes []route.RouteImpl, tokenIn sdk.Coin, maxRoutes int, logger log.Logger) (domain.Quote, []route.RouteImpl, error) {
-	topQuote, routesSortedByAmtOut, err := estimateAndRankSingleRouteQuote(ctx, routes, tokenIn, logger)
-	if err != nil {
-		return nil, nil, err
-	}
-
-	numRoutes := len(routesSortedByAmtOut)
-
-	// If split routes are disabled, return a single the top route
-	if maxRoutes == 0 && numRoutes > 0 {
-		numRoutes = 1
-		// If there are more routes than the max split routes, keep only the top routes
-	} else if len(routesSortedByAmtOut) > maxRoutes {
-		// Keep only top routes for splits
-		routes = routes[:maxRoutes]
-		numRoutes = maxRoutes
-	}
-
-	// Convert routes sorted by amount out to routes
-	for i := 0; i < numRoutes; i++ {
-		// Update routes with the top routes
-		routes[i] = routesSortedByAmtOut[i].RouteImpl
-	}
```

Comment:
Note to reviewer: this logic was moved to a different level of abstraction and refactored a bit for testability and to avoid redundant iterations